diff -ru --strip-trailing-cr cd.orig/tecmake.mak cd/tecmake.mak
--- cd.orig/tecmake.mak	2010-11-12 18:04:08.000000000 +0100
+++ cd/tecmake.mak	2010-12-02 18:36:43.294100800 +0100
@@ -921,6 +921,8 @@
     # get compile/link flags via pkg-config
     STDFLAGS += $(shell pkg-config --cflags gtk+-2.0 gdk-2.0)
     LIBS += $(shell pkg-config --libs gtk+-2.0 gdk-2.0)
+    GTK_BASE := $(shell pkg-config --variable=prefix gtk+-2.0)
+    GTK := $(GTK_BASE)
   else
     CHECK_GTK = Yes
     ifneq ($(findstring MacOS, $(TEC_UNAME)), )
