diff -ru --strip-trailing-cr im.orig/src/libexif/config.h im/src/libexif/config.h
--- im.orig/src/libexif/config.h	2015-02-05 16:54:16.956068700 +0100
+++ im/src/libexif/config.h	2015-02-06 18:52:07.037795100 +0100
@@ -6,7 +6,7 @@
 /* The gettext domain we're using */
 #define GETTEXT_PACKAGE "libexif-9"
 
-#ifdef WIN32
+#if defined(WIN32) && !defined(__CYGWIN__)
 #define snprintf _snprintf
 #endif
 
