diff -ru --strip-trailing-cr im.orig/src/liblzf/lzf_c.c im/src/liblzf/lzf_c.c
--- im.orig/src/liblzf/lzf_c.c	2009-05-01 02:26:38.000000000 +0200
+++ im/src/liblzf/lzf_c.c	2010-11-19 08:43:37.799002500 +0100
@@ -121,7 +121,7 @@
    * special workaround for it.
    */
 #if defined (WIN32) && defined (_M_X64)
-  unsigned _int64 off; /* workaround for missing POSIX compliance */
+  long long off; /* workaround for missing POSIX compliance */
 #else
   unsigned long off;
 #endif
