diff -ru --strip-trailing-cr im.orig/tecmake.mak im/tecmake.mak
--- im.orig/tecmake.mak	2010-11-12 18:04:08.000000000 +0100
+++ im/tecmake.mak	2010-12-07 20:51:19.777521800 +0100
@@ -919,7 +919,7 @@
 ifdef USE_GTK
   ifdef USE_PKGCONFIG
     # get compile/link flags via pkg-config
-    STDFLAGS += $(shell pkg-config --cflags gtk+-2.0 gdk-2.0)
+    EXTRAINCS += $(shell pkg-config --cflags gtk+-2.0 gdk-2.0)
     LIBS += $(shell pkg-config --libs gtk+-2.0 gdk-2.0)
   else
     CHECK_GTK = Yes
