diff -ru im-3.6.3/tecmake.mak im-3.6.3.patched/tecmake.mak
--- im-3.6.3/tecmake.mak	2010-11-12 18:04:08.000000000 +0100
+++ im-3.6.3.patched/tecmake.mak	2011-07-28 20:59:26.412564600 +0200
@@ -46,6 +46,10 @@
     TEC_SYSMINOR:=$(shell sw_vers -productVersion|cut -f2 -d.)
     TEC_SYSARCH:=$(shell uname -p)
   endif
+  ifeq ($(TEC_SYSNAME), GNU/kFreeBSD)
+    TEC_SYSNAME:=kFreeBSD
+    TEC_SYSMINOR:=$(shell uname -r|cut -f2 -d.|cut -f1 -d-)
+  endif
 
   ifeq ($(TEC_SYSARCH), i686)
     TEC_SYSARCH:=x86
