diff -ru --strip-trailing-cr im.orig/src/Makefile im/src/Makefile
--- im.orig/src/Makefile	2010-04-23 14:30:22.000000000 +0200
+++ im/src/Makefile	2010-08-23 21:44:01.968924500 +0200
@@ -1,7 +1,6 @@
-
-.PHONY: do_all im im_jp2 im_process im_fftw imlua3 imlua5 imlua_jp2 imlua_process5 imlua_fftw5
+.PHONY: do_all im im_jp2 im_process im_fftw im_ecw im_capture im_avi im_wmv im_fftw3 imlua3 imlua5 imlua_jp2 imlua_process5 imlua_fftw5
 #do_all: im im_jp2 im_process im_fftw imlua3 imlua5 imlua_jp2 imlua_process5 imlua_fftw5
-do_all: im im_jp2 im_process im_fftw imlua5 imlua_jp2 imlua_process5 imlua_fftw5
+do_all: im im_jp2 im_process im_fftw im_ecw im_capture im_avi im_wmv im_fftw3 imlua3 imlua5 imlua_jp2 imlua_process5 imlua_fftw5
 
 im:
 	$(MAKE) --no-print-directory -f tecmake.mak
@@ -21,3 +20,13 @@
 	@$(MAKE) --no-print-directory -f tecmake.mak MF=imlua_process5
 imlua_fftw5:
 	@$(MAKE) --no-print-directory -f tecmake.mak MF=imlua_fftw5
+im_avi:
+	@$(MAKE) --no-print-directory -f tecmake.mak MF=im_avi
+im_capture:
+	@$(MAKE) --no-print-directory -f tecmake.mak MF=im_capture
+im_wmv:
+	@$(MAKE) --no-print-directory -f tecmake.mak MF=im_wmv
+im_fftw3:
+	@$(MAKE) --no-print-directory -f tecmake.mak MF=im_fftw3
+im_ecw:
+	@$(MAKE) --no-print-directory -f tecmake.mak MF=im_ecw
diff -ru --strip-trailing-cr im.orig/src/config.mak im/src/config.mak
--- im.orig/src/config.mak	2010-06-23 22:56:34.000000000 +0200
+++ im/src/config.mak	2010-08-23 21:44:01.968924500 +0200
@@ -114,7 +114,11 @@
       USE_EXIF = Yes
     endif
 else
-  SRC += im_sysfile_unix.cpp
+  ifneq ($(findstring cygw, $(TEC_SYSNAME)), )
+    SRC += im_sysfile_unix.cpp im_dib.cpp im_dibxbitmap.cpp
+  else
+    SRC += im_sysfile_unix.cpp
+  endif
 endif
 
 ifdef USE_EXIF
diff -ru --strip-trailing-cr im.orig/src/im_capture.mak im/src/im_capture.mak
--- im.orig/src/im_capture.mak	2010-06-22 19:47:20.000000000 +0200
+++ im/src/im_capture.mak	2010-08-23 21:44:01.968924500 +0200
@@ -51,13 +51,14 @@
 ifneq ($(findstring Win, $(TEC_SYSNAME)), )
   INCLUDES += $(DXSDK)/include
   SRC = im_capture_dx.cpp
+  LIBS = oleaut32 uuid
 endif
 
 #ifneq ($(findstring Linux, $(TEC_UNAME)), )
 #  SRC = im_capture_v4l.cpp
 #endif
              
-LIBS = strmiids
+LIBS += strmiids
 
 mingw3-dll:                    
 	@echo Importing MingW stub library
diff -ru --strip-trailing-cr im.orig/src/tecmake.mak im/src/tecmake.mak
--- im.orig/src/tecmake.mak	2010-06-23 21:23:20.000000000 +0200
+++ im/src/tecmake.mak	2010-08-23 21:45:13.813664700 +0200
@@ -8,9 +8,8 @@
 # Tecmake Version
 VERSION = 4.1
 
-
 #---------------------------------#
-# First target 
+# First target
 .PHONY: build
 build: tecmake
 
@@ -59,11 +58,36 @@
   ifeq ($(TEC_SYSARCH), x86_64)
     TEC_SYSARCH:=x64
   endif
+  ifeq ($(TEC_SYSARCH), amd64)
+    TEC_SYSARCH:=x64
+  endif
 
   # Compose
   TEC_SYSRELEASE:=$(TEC_SYSVERSION).$(TEC_SYSMINOR)
   TEC_UNAME:=$(TEC_SYSNAME)$(TEC_SYSVERSION)$(TEC_SYSMINOR)
 
+  # Cygwin
+  ifneq ($(findstring CYGWIN, $(TEC_SYSNAME)), )
+    TEC_SYSNAME:=CYGWIN
+    TEC_UNAME:=cygw$(TEC_SYSVERSION)$(TEC_SYSMINOR)
+  endif
+
+  # Mingw + MSYS
+  ifneq ($(findstring MINGW, $(TEC_SYSNAME)), )
+    TEC_SYSVERSION:=$(shell uname -s|cut -f2 -d-|cut -f1 -d.)
+    TEC_SYSMINOR:=$(shell uname -s|cut -f2 -d-|cut -f2 -d.)
+    TEC_SYSRELEASE:=$(TEC_SYSVERSION).$(TEC_SYSMINOR)
+    ifeq ($(shell gcc -dumpmachine|cut -f1 -d-), x86_64)
+      TEC_SYSARCH:=x64
+      TEC_SYSNAME:=Win64
+    else
+      TEC_SYSARCH:=x86
+      TEC_SYSNAME:=Win32
+    endif
+    GCCVER:=$(shell gcc -dumpversion|cut -f1 -d.)
+    TEC_UNAME:=mingw$(GCCVER)
+  endif
+
   # Linux 2.4 and GCC 3.x
   ifeq ($(TEC_UNAME), Linux24)
     GCCVER:=$(shell gcc -dumpversion|cut -f1 -d.)
@@ -88,14 +112,16 @@
   endif
 
   # 64-bits Linux
-  ifeq ($(TEC_SYSARCH), x64)
-    BUILD_64=Yes
-    TEC_UNAME:=$(TEC_UNAME)_64
-  endif
+  ifeq ($(TEC_SYSNAME), Linux)
+    ifeq ($(TEC_SYSARCH), x64)
+      BUILD_64=Yes
+      TEC_UNAME:=$(TEC_UNAME)_64
+    endif
 
-  ifeq ($(TEC_SYSARCH), ia64)
-    BUILD_64=Yes
-    TEC_UNAME:=$(TEC_UNAME)_ia64
+    ifeq ($(TEC_SYSARCH), ia64)
+      BUILD_64=Yes
+      TEC_UNAME:=$(TEC_UNAME)_ia64
+    endif
   endif
 
   # Solaris and Intel
@@ -127,13 +153,31 @@
 	@echo 'TEC_SYSVERSION = $(TEC_SYSVERSION)'
 	@echo 'TEC_SYSMINOR = $(TEC_SYSMINOR)'
 	@echo 'TEC_SYSARCH = $(TEC_SYSARCH)'
-	@echo 'TEC_UNAME = $(TEC_UNAME)'; echo ''
+	@echo 'TEC_UNAME = $(TEC_UNAME)'
+	@echo 'USE_X11 = $(USE_X11)'
+	@echo 'USE_MOT = $(USE_MOT)'
+	@echo 'USE_GTK = $(USE_GTK)'
+	@echo 'USE_GDK = $(USE_GDK)'
+	@echo 'USE_WIN = $(USE_WIN)'
+	@echo 'GTK = $(GTK)'
+	@echo 'X11_LIBS = $(X11_LIBS)'
+	@echo 'X11_LIB = $(X11_LIB)'
+	@echo 'X11_INC = $(X11_INC)'
+	@echo 'MOTIFGL_LIB = $(MOTIFGL_LIB)'
+	@echo 'MOTIF_LIB = $(MOTIF_LIB)'
+	@echo 'MOTIF_INC = $(MOTIF_INC)'
+	@echo 'GLUT_LIB = $(GLUT_LIB)'
+	@echo 'GLUT_INC = $(GLUT_INC)'
+	@echo 'OPENGL_LIBS = $(OPENGL_LIBS)'
+	@echo 'OPENGL_LIB = $(OPENGL_LIB)'
+	@echo 'OPENGL_INC = $(OPENGL_INC)'
+	@echo 'LIBS = $(LIBS)'; echo ''
+
 
-  
 #---------------------------------#
 # Known Platforms
 
-UNAMES = Linux24 Linux24g3 Linux24g3_64 Linux26 Linux26_64 Linux26g4 Linux26g4_64 Linux26_ia64 FreeBSD54 SunOS57 SunOS58 SunOS510 SunOS510_x86 AIX43 IRIX65 IRIX6465
+UNAMES = Linux24 Linux24g3 Linux24g3_64 Linux26 Linux26_64 Linux26g4 Linux26g4_64 Linux26_ia64 FreeBSD54 SunOS57 SunOS58 SunOS510 SunOS510_x86 AIX43 IRIX65 IRIX6465 mingw3 mingw4 cygw15 cygw17
 
 
 #---------------------------------#
@@ -179,32 +223,96 @@
 # Compilation Flags
 STDFLAGS := -Wall
 STDDEFS  := -DTEC_UNAME=$(TEC_UNAME) -DTEC_SYSNAME=$(TEC_SYSNAME) -D$(TEC_SYSNAME)=$(TEC_SYSRELEASE) -D$(TEC_BYTEORDER) -D$(TEC_WORDSIZE) -DFUNCPROTO=15
-STDINCS  := 
+STDINCS  :=
 OPTFLAGS := -O2
 STDLFLAGS  := r
-DEBUGFLAGS := -g  
+DEBUGFLAGS := -g
 STDLDFLAGS := -shared
 DLIBEXT := so
 DLIBPRE := lib
 APPEXT :=
 
 
+ifneq ($(findstring mingw, $(TEC_UNAME)), )
+  WIN_DEFAULT = Yes
+endif
 ifneq ($(findstring cygw, $(TEC_UNAME)), )
   GTK_DEFAULT = Yes
-endif  
+endif
 ifneq ($(findstring Linux, $(TEC_UNAME)), )
   GTK_DEFAULT = Yes
-endif  
+endif
 ifneq ($(findstring MacOS, $(TEC_UNAME)), )
   GTK_DEFAULT = Yes
-endif  
+endif
 ifneq ($(findstring FreeBSD, $(TEC_UNAME)), )
   GTK_DEFAULT = Yes
-endif  
+endif
 ifneq ($(findstring Linux24, $(TEC_UNAME)), )
   GTK_DEFAULT :=
-endif  
+endif
+
+ifndef USE_X11
+  ifndef USE_MOTIF
+    ifndef USE_WIN
+      ifndef USE_GTK
+        ifndef USE_GDK
+          ifdef GTK_DEFAULT
+            override USE_GTK := Yes
+          endif
+          ifdef WIN_DEFAULT
+            override USE_WIN := Yes
+          endif
+	endif
+      endif
+    endif
+  endif
+endif
+
+ifdef USE_GTK
+  override USE_GDK := Yes
+  override USE_MOTIF :=
+  override USE_X11 :=
+  override USE_WIN :=
+  override IUPSUFX := gtk
+  override CDSUFX := gdk
+endif
+
+ifdef USE_GDK
+  override USE_GTK := Yes
+  override USE_MOTIF :=
+  override USE_X11 :=
+  override USE_WIN :=
+  override IUPSUFX := gtk
+  override CDSUFX := gdk
+endif
+
+ifdef USE_X11
+  override USE_MOTIF := Yes
+  override USE_GTK :=
+  override USE_GDK :=
+  override USE_WIN :=
+  override IUPSUFX := mot
+  override CDSUFX := x11
+endif
+
+ifdef USE_MOTIF
+  override USE_X11 := Yes
+  override USE_GTK :=
+  override USE_GDK :=
+  override USE_WIN :=
+  override IUPSUFX := mot
+  override CDSUFX := x11
+endif
 
+ifdef USE_WIN
+  override USE_X11 :=
+  override USE_MOTIF :=
+  override USE_GTK :=
+  override USE_GDK :=
+  override IUPSUFX := win
+  override CDSUFX := win
+endif
 
 #---------------------------------#
 # Tools
@@ -215,7 +323,7 @@
 RANLIB   := $(TEC_TOOLCHAIN)ranlib
 AR       := $(TEC_TOOLCHAIN)ar
 DEBUGGER := $(TEC_TOOLCHAIN)gdb
-RCC      := $(TEC_TOOLCHAIN)windres 
+RCC      := $(TEC_TOOLCHAIN)windres
 
 # Remote build script
 REMOTE  = $(TECMAKE_HOME)/remote
@@ -239,7 +347,7 @@
 
 
 #---------------------------------#
-# Definitions of public variables 
+# Definitions of public variables
 
 ifdef LIBNAME
   TARGETNAME = $(LIBNAME)
@@ -250,7 +358,7 @@
 endif
 
 ifndef TARGETNAME
-  $(error LIBNAME nor APPNAME defined in $(MAKENAME)) 
+  $(error LIBNAME nor APPNAME defined in $(MAKENAME))
 endif
 
 PROJNAME ?= $(TARGETNAME)
@@ -285,7 +393,7 @@
 ifdef BUILD_64
   ifneq ($(findstring SunOS, $(TEC_UNAME)), )
     USE_CC = Yes
-    BUILD_64_DIR = Yes   
+    BUILD_64_DIR = Yes
   endif
   ifneq ($(findstring AIX, $(TEC_UNAME)), )
     USE_CC = Yes
@@ -296,11 +404,11 @@
     BUILD_64_DIR = Yes
   endif
 endif
-  
+
 ifdef USE_CC
   CC := cc
   CPPC := CC
-  STDFLAGS = 
+  STDFLAGS =
   UNAMES := $(UNAMES_CC)
   ifdef USE_CC_DIR
     TEC_UNAME := $(TEC_UNAME)cc
@@ -387,22 +495,22 @@
 
 # Definitions for X11
 X11_LIBS := Xmu Xt Xext X11
-#X11_LIB := 
+#X11_LIB :=
 #X11_INC :=                     #include <X11/X.h>
 
 # Definitions for OpenGL
 OPENGL_LIBS := GLU GL
-#OPENGL_LIB := 
-#OPENGL_INC :=                  #include <GL/gl.h>  and possibly  
+#OPENGL_LIB :=
+#OPENGL_INC :=                  #include <GL/gl.h>  and possibly
 MOTIFGL_LIB := GLw              #include <GL/GLwMDrawA.h>
 
 # Definitions for Motif
-#MOTIF_LIB := 
+#MOTIF_LIB :=
 #MOTIF_INC :=                   #include <Xm/Xm.h>
 
 # Definitions for GLUT
-#GLUT_LIB := 
-#GLUT_INC := 
+#GLUT_LIB :=
+#GLUT_INC :=
 
 # Definitions for GTK
 ifdef GTK_BASE
@@ -419,6 +527,7 @@
 endif
 
 ifneq ($(findstring Linux, $(TEC_UNAME)), )
+  UNIX_LINUX = Yes
   ifdef BUILD_64
     ifeq ($(TEC_SYSARCH), ia64)
       STDFLAGS += -fPIC
@@ -431,16 +540,17 @@
     X11_LIB := /usr/X11R6/lib
   endif
   X11_INC := /usr/X11R6/include
-  MOTIFGL_LIB := 
+  MOTIFGL_LIB :=
 endif
 
 ifneq ($(findstring IRIX, $(TEC_UNAME)), )
+  UNIX_POSIX = Yes
   LD = ld
   STDLDFLAGS := -elf -shared -rdata_shared -soname lib$(TARGETNAME).so
   RANLIB := /bin/true
   X11_LIBS := Xmu Xt X11
-  ifdef BUILD_64    
-    ifdef USE_CC  
+  ifdef BUILD_64
+    ifdef USE_CC
       STDFLAGS += -64 -KPIC
       STDLDFLAGS += -64
       LINKER += -64
@@ -454,10 +564,11 @@
   MOTIF_INC = /usr/Motif-2.1/include
 endif
 
-ifneq ($(findstring AIX, $(TEC_UNAME)), ) 
+ifneq ($(findstring AIX, $(TEC_UNAME)), )
+  UNIX_POSIX = Yes
   NO_DYNAMIC ?= Yes
   ifdef BUILD_64
-    ifdef USE_CC  
+    ifdef USE_CC
       STDFLAGS += -q64 # to compilers C and C++
       STDLFLAGS := -X64 $(STDLFLAGS) # to librarian
       STDLDFLAGS += -64
@@ -467,6 +578,7 @@
 endif
 
 ifneq ($(findstring HP-UX, $(TEC_UNAME)), )
+  UNIX_POSIX = Yes
   NO_DYNAMIC ?= Yes
   MOTIF_INC := /usr/include/Motif2.1
   X11_LIBS := Xt Xext X11
@@ -479,6 +591,7 @@
 endif
 
 ifneq ($(findstring SunOS, $(TEC_UNAME)), )
+  UNIX_POSIX = Yes
   LD = ld
   STDLDFLAGS := -G
   X11_INC := /usr/openwin/share/include
@@ -489,16 +602,17 @@
   GLUT_LIB := /usr/local/glut-3.7/lib/glut
   GLUT_INC := /usr/local/glut-3.7/include
   ifdef BUILD_64
-    ifdef USE_CC  
+    ifdef USE_CC
       STDFLAGS += -xarch=v9 -KPIC
       # have to force these PATHs because of a conflict with standard PATHs
-      STDLDFLAGS += -64 -L/usr/lib/64 -L/usr/ucblib/sparcv9  
+      STDLDFLAGS += -64 -L/usr/lib/64 -L/usr/ucblib/sparcv9
       LINKER += -xarch=v9
     endif
   endif
 endif
 
 ifneq ($(findstring MacOS, $(TEC_UNAME)), )
+  UNIX_BSD = Yes
   X11_LIBS := Xmu Xp Xt Xext X11
   X11_LIB := /usr/X11R6/lib
   X11_INC := /usr/X11R6/include
@@ -511,11 +625,44 @@
   else
     STDLDFLAGS := -bundle -undefined dynamic_lookup
   endif
+  ifdef USE_OPENGL
+    ifeq ($(TEC_SYSMINOR), 5)
+      #Darwin9 Only - OpenGL bug fix for Fink, when the message bellow appears
+      #   ld: cycle in dylib re-exports with /usr/X11R6/lib/libGL.dylib
+      LFLAGS += -dylib_file /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib:/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib
+    endif
+  endif
 endif
 
 ifneq ($(findstring FreeBSD, $(TEC_UNAME)), )
+  BSD = Yes
   X11_LIB := /usr/X11R6/lib
   X11_INC := /usr/X11R6/include
+  ifeq ($(TEC_SYSARCH), x64)
+    STDFLAGS += -fPIC
+  endif
+endif
+
+ifneq ($(findstring cygw, $(TEC_UNAME)), )
+  ifndef USE_WIN
+    OPENGL_LIBS := GLU GL
+    X11_LIBS := X11 Xt
+  else
+    #we are gonna use native windows opengl32
+    OPENGL_LIBS := opengl32 glu32
+    X11_LIBS :=
+    DEFINES += USE_OPENGL32 WIN32
+  endif
+  MOTIFGL_LIB :=
+endif
+
+ifneq ($(findstring Win, $(TEC_SYSNAME)), )
+  WIN_OTHER := Yes
+  DLIBEXT := dll
+  APPEXT := .exe
+  STDLDFLAGS += -Wl,--out-implib=$(TARGETDIR)/lib$(TARGETNAME).dll.a
+  OPENGL_LIBS := opengl32 glu32
+  X11_LIBS :=
 endif
 
 #---------------------------------#
@@ -589,23 +736,23 @@
 
 ifdef USE_IUP3
   override USE_IUP = Yes
-# Inside Tecgraf only  
+# Inside Tecgraf only
   ifndef IUP3_BUILD
-#  IUP := $(IUP)3
+#    IUP := $(IUP)3
   endif
-endif 
+endif
 
 ifdef USE_IUPBETA
   IUP := $(IUP)/beta
-endif 
+endif
 
 ifdef USE_CDBETA
   CD := $(CD)/beta
-endif 
+endif
 
 ifdef USE_IMBETA
   IM := $(IM)/beta
-endif 
+endif
 
 ifdef USE_GLUT
   override USE_OPENGL = Yes
@@ -632,7 +779,7 @@
     LIBS += iupcontrols
   endif
 endif
-  
+
 ifdef USE_IMLUA
   override USE_IM = Yes
   ifdef USE_STATIC
@@ -693,39 +840,24 @@
       endif
     endif
   endif
-  
+
   LUA_INC   ?= $(LUA)/include
   INCLUDES += $(LUA_INC)
-  
+
   LUA_BIN ?= $(LUA)/bin/$(TEC_UNAME)
   BIN2C     := $(LUA_BIN)/bin2c$(LUASUFX)
   LUAC      := $(LUA_BIN)/luac$(LUASUFX)
   LUABIN    := $(LUA_BIN)/lua$(LUASUFX)
 endif
 
-ifdef USE_IUP   
-  IUPSUFX := 
-  ifdef USE_IUP3
-    ifdef GTK_DEFAULT
-      ifdef USE_MOTIF
-        IUPSUFX := mot
-      else
-        override USE_GTK = Yes
-        override USE_GDK = Yes
-      endif
-    else
-      ifdef USE_GTK
-        IUPSUFX := gtk
-      else
-        override USE_MOTIF = Yes
-      endif
-    endif
-  else
+ifdef USE_IUP
+  ifndef USE_IUP3
+    override IUPSUFX :=
     override USE_MOTIF = Yes
   endif
   ifdef USE_STATIC
     ifdef USE_CD
-      SLIB += $(IUP)/lib/$(TEC_UNAME_LIB_DIR)/libiupcd.a
+      SLIB += $(IUP)/lib/$(TEC_UNAME_LIB_DIR)/libiupcd$(CDSUFFIX).a
     endif
     ifdef USE_OPENGL
       SLIB += $(IUP)/lib/$(TEC_UNAME_LIB_DIR)/libiupgl.a
@@ -733,7 +865,7 @@
     SLIB += $(IUP)/lib/$(TEC_UNAME_LIB_DIR)/libiup$(IUPSUFX).a
   else
     ifdef USE_CD
-      LIBS += iupcd
+      LIBS += iupcd$(CDSUFX)
     endif
     ifdef USE_OPENGL
       LIBS += iupgl
@@ -745,21 +877,12 @@
 endif
 
 ifdef USE_CD
-  CDSUFX := 
-  override USE_X11 = Yes
-  ifdef USE_IUP3
-    ifdef USE_GDK
-      ifndef GTK_DEFAULT
-        CDSUFX := gdk
-      endif
-    else
-      ifdef GTK_DEFAULT
-        CDSUFX := x11
-      endif
-    endif
+  ifdef USE_CD_OLD
+    override CDSUFX :=
   endif
   ifdef USE_STATIC
     ifdef USE_XRENDER
+      CHECK_XRENDER = Yes
       SLIB += $(CD)/lib/$(TEC_UNAME_LIB_DIR)/libcdcontextplus.a
       LIBS += Xrender Xft
     endif
@@ -776,6 +899,7 @@
     endif
   else
     ifdef USE_XRENDER
+      CHECK_XRENDER = Yes
       LIBS += cdcontextplus
       LIBS += Xrender Xft
     endif
@@ -789,7 +913,15 @@
     LDIR += $(CD)/lib/$(TEC_UNAME_LIB_DIR)
     ifndef USE_GTK
       # Freetype is already included in GTK
-      LIBS += freetype
+      ifneq ($(findstring cygw, $(TEC_UNAME)), )
+        LIBS += freetype-6
+      else
+        ifneq ($(findstring Win, $(TEC_SYSNAME)), )
+          LIBS += freetype6
+        else
+          LIBS += freetype
+        endif
+      endif
     endif
   endif
   INCLUDES += $(CD)/include
@@ -809,7 +941,7 @@
   LIBS += glut
   LDIR += $(GLUT_LIB)
   STDINCS += $(GLUT_INC)
-endif 
+endif
 
 ifdef USE_OPENGL
   override USE_X11 = Yes
@@ -821,22 +953,22 @@
   LIBS += $(OPENGL_LIBS)
   LDIR += $(OPENGL_LIB)
   STDINCS += $(OPENGL_INC)
-endif 
+endif
 
 ifdef USE_MOTIF
-  override USE_X11 = Yes
   LIBS += Xm
   LDIR += $(MOTIF_LIB)
   STDINCS += $(MOTIF_INC)
   ifneq ($(findstring Linux, $(TEC_UNAME)), )
     X11_LIBS := Xpm $(X11_LIBS)
   endif
-  ifneq ($(findstring cygw, $(TEC_UNAME)), ) 
+  ifneq ($(findstring cygw, $(TEC_UNAME)), )
     X11_LIBS := Xpm $(X11_LIBS)
   endif
 endif
 
 ifdef USE_GTK
+  CHECK_GTK = Yes
   ifneq ($(findstring MacOS, $(TEC_UNAME)), )
 # Option 1 - Fink GTK port
     LDIR += $(GTK)/lib
@@ -862,10 +994,10 @@
     override USE_X11 = Yes
     LIBS += gtk-x11-2.0 gdk-x11-2.0 pangox-1.0
   endif
-  
+
   LIBS += gdk_pixbuf-2.0 pango-1.0 gobject-2.0 gmodule-2.0 glib-2.0
   STDINCS += $(GTK)/include/atk-1.0 $(GTK)/include/gtk-2.0 $(GTK)/include/cairo $(GTK)/include/pango-1.0 $(GTK)/include/glib-2.0
-  
+
   ifeq ($(TEC_SYSARCH), x64)
     STDINCS += $(GTK)/lib64/glib-2.0/include $(GTK)/lib64/gtk-2.0/include
     # Add also these to avoid errors in systems that lib64 does not exists
@@ -878,7 +1010,7 @@
   endif
   endif
   ifneq ($(findstring FreeBSD, $(TEC_UNAME)), )
-    STDINCS += /lib/X11R6/include/gtk-2.0  
+    STDINCS += /lib/X11R6/include/gtk-2.0
   endif
 endif
 
@@ -894,24 +1026,31 @@
   LIBS += $(X11_LIBS)
   LDIR += $(X11_LIB)
   STDINCS += $(X11_INC)
-endif 
+endif
+
+ifdef USE_WIN
+  LIBS += gdi32 comctl32 comdlg32 ole32
+  DEFINES += IUP_DLL
+endif
 
 LIBS += m
 
 ifneq ($(findstring cygw, $(TEC_UNAME)), )
+  WIN_OTHER := Yes
+
   # INCLUDES for dependencies, remove references to "c:" and similars
   DEPINCS := $(patsubst c:%, /cygdrive/c%, $(INCLUDES))
   DEPINCS := $(patsubst d:%, /cygdrive/d%, $(DEPINCS))
   DEPINCS := $(patsubst x:%, /cygdrive/x%, $(DEPINCS))
   DEPINCS := $(patsubst t:%, /cygdrive/t%, $(DEPINCS))
-  
+
   DLIBEXT := dll
   APPEXT := .exe
-  # Use the cyg prefix to indicate that it is a Cygwin Posix DLL  
+  # Use the cyg prefix to indicate that it is a Cygwin Posix DLL
   DLIBPRE := cyg
-  
+
   STDLDFLAGS += -Wl,--out-implib=$(TARGETDIR)/lib$(TARGETNAME).dll.a
-endif 
+endif
 
 #---------------------------------#
 #  Building compilation flags that are sets
@@ -951,10 +1090,10 @@
 ifeq ($(MAKETYPE), APP)
   TARGET := $(TARGETDIR)/$(TARGETAPPNAME)
 else
-  ifeq ($(NO_DYNAMIC), Yes) 
+  ifeq ($(NO_DYNAMIC), Yes)
     TARGET := $(TARGETDIR)/$(TARGETSLIBNAME)
   else
-  ifeq ($(NO_STATIC), Yes) 
+  ifeq ($(NO_STATIC), Yes)
     TARGET := $(TARGETDIR)/$(TARGETDLIBNAME)
   else
     TARGET := $(TARGETDIR)/$(TARGETSLIBNAME) $(TARGETDIR)/$(TARGETDLIBNAME)
@@ -999,17 +1138,17 @@
 #---------------------------------#
 # Main Rule - Build Everything that it is necessary
 
-.PHONY: tecmake 
+.PHONY: tecmake
 ifeq ($(MAKETYPE), APP)
-  tecmake: print-start directories application scripts
+  tecmake: print-start system-check directories application scripts
 else
-  ifeq ($(NO_DYNAMIC), Yes) 
-    tecmake: print-start directories static-lib
+  ifeq ($(NO_DYNAMIC), Yes)
+    tecmake: print-start system-check directories static-lib
   else
-  ifeq ($(NO_STATIC), Yes) 
-    tecmake: print-start directories dynamic-lib
+  ifeq ($(NO_STATIC), Yes)
+    tecmake: print-start system-check directories dynamic-lib
   else
-    tecmake: print-start directories static-lib dynamic-lib
+    tecmake: print-start system-check directories static-lib dynamic-lib
   endif
   endif
 endif
@@ -1018,7 +1157,30 @@
 print-start:
 	@echo ''; echo 'Tecmake: starting [ $(TARGETNAME):$(TEC_UNAME) ]'
 
-  
+.PHONY: system-check
+system-check:
+  ifdef CHECK_XRENDER
+    ifdef UNIX_POSIX
+			@echo ''; echo 'Tecmake: check failed, XRender NOT available in this system.'; echo ''; exit 1;
+    endif
+  endif
+  ifdef CHECK_GTK
+    ifndef GTK_BASE
+      ifdef UNIX_POSIX
+		@echo ''; echo 'Tecmake: check failed, GTK NOT available in this system.'; echo ''; exit 1;
+      else
+        ifneq ($(findstring Linux24, $(TEC_UNAME)), )
+					@echo ''; echo 'Tecmake: check failed, GTK too OLD in this system.'; echo ''; exit 1;
+        endif
+      endif
+    endif
+  endif
+  ifdef CHECK_GDIPLUS
+    ifdef WIN_OTHER
+			@echo ''; echo 'Tecmake: check failed, GDI+ NOT available in this system.'; echo ''; exit 1;
+    endif
+  endif
+
 #---------------------------------#
 # Dynamic Library Build
 
@@ -1026,11 +1188,11 @@
 dynamic-lib: $(TARGETDIR)/$(TARGETDLIBNAME)
 
 $(TARGETDIR)/$(TARGETDLIBNAME) : $(LOHS) $(OBJS) $(EXTRADEPS)
-	@echo ''; echo Tecmake: linking $(<F)...
+	@echo ''; echo Tecmake: linking $(@F) ...
 	$(ECHO)$(LD) $(STDLDFLAGS) -o $@ $(OBJS) $(SLIB) $(LFLAGS)
 	@echo ''; echo 'Tecmake: Dynamic Library ($@) Done.'; echo ''
 
-  
+
 #---------------------------------#
 # Static Library Build
 
@@ -1038,13 +1200,13 @@
 static-lib: $(TARGETDIR)/$(TARGETSLIBNAME)
 
 $(TARGETDIR)/$(TARGETSLIBNAME) : $(LOHS) $(OBJS) $(EXTRADEPS)
-	@echo ''; echo Tecmake: librarian $(<F)...
+	@echo ''; echo Tecmake: librarian $(@F) ...
 	$(ECHO)$(AR) $(STDLFLAGS) $@ $(OBJS) $(SLIB) $(LCFLAGS)
-	@echo ''; echo Tecmake: updating lib TOC $(<F)...
+	@echo ''; echo Tecmake: updating lib TOC $(@F) ...
 	$(ECHO)-$(RANLIB) $@
 	@echo ''; echo 'Tecmake: Static Library ($@) Done.'; echo ''
 
-  
+
 #---------------------------------#
 # Application Build
 
@@ -1052,15 +1214,15 @@
 application: $(TARGETDIR)/$(TARGETAPPNAME)
 
 $(TARGETDIR)/$(TARGETAPPNAME) : $(LOHS) $(OBJS) $(EXTRADEPS)
-	@echo ''; echo Tecmake: linking $(<F)...
+	@echo ''; echo Tecmake: linking $(@F) ...
 	$(ECHO)$(LINKER) -o $@ $(OBJS) $(SLIB) $(LFLAGS)
 	@if [ ! -z "$(STRIP)" ]; then \
 	   echo ''; echo 'Tecmake: striping debug information' ;\
-	   $(ECHO)strip $@ ;\
+	   strip $@ ;\
 	 fi
 	@echo ''; echo 'Tecmake: Application ($@) Done.'; echo ''
 
-  
+
 #---------------------------------#
 #  Application Scripts
 
@@ -1073,7 +1235,7 @@
 else
   scripts: $(SRELEASE) ;
 endif
-  
+
 $(SRELEASE): $(MAKENAME)
 	@echo ''; echo 'Tecmake: building script $(@F)'
 	@echo "#!/bin/csh" > $@
@@ -1084,7 +1246,7 @@
 	@echo 'exec $(TARGETROOT)/$$TEC_UNAME/$(TARGETNAME) $$*' >> $@
 	@chmod a+x $@
 
-  
+
 #---------------------------------#
 # Directories Creation
 
@@ -1113,52 +1275,51 @@
 # Compilation Rules
 
 $(OBJDIR)/%.o:  $(SRCDIR)/%.c
-	@echo ''; echo Tecmake: compiling $(<F)...
+	@echo ''; echo Tecmake: compiling $(<F) ...
 	$(ECHO)$(CC) -c $(CFLAGS) -o $@ $<
 
 $(OBJDIR)/%.o:  $(SRCDIR)/%.cpp
-	@echo ''; echo Tecmake: compiling $(<F)...
+	@echo ''; echo Tecmake: compiling $(<F) ...
 	$(ECHO)$(CPPC) -c $(CXXFLAGS) -o $@ $<
 
 $(OBJDIR)/%.o:  $(SRCDIR)/%.cxx
-	@echo ''; echo Tecmake: compiling $(<F)...
+	@echo ''; echo Tecmake: compiling $(<F) ...
 	$(ECHO)$(CPPC) -c $(CXXFLAGS) -o $@ $<
 
 $(OBJDIR)/%.o:  $(SRCDIR)/%.cc
-	@echo ''; echo Tecmake: compiling $(<F)...
+	@echo ''; echo Tecmake: compiling $(<F) ...
 	$(ECHO)$(CPPC) -c $(CXXFLAGS) -o $@ $<
 
 $(OBJDIR)/%.o: $(SRCDIR)/%.f
-	@echo ''; echo Tecmake: compiling $(<F)...
+	@echo ''; echo Tecmake: compiling $(<F) ...
 	$(ECHO)$(FC) -c $(FFLAGS) -o $@ $<
 
 $(OBJDIR)/%.o: $(SRCDIR)/%.for
-	@echo ''; echo Tecmake: compiling $(<F)...
+	@echo ''; echo Tecmake: compiling $(<F) ...
 	$(ECHO)$(FC) -c $(FFLAGS) -o $@ $<
 
 $(OBJDIR)/%.ro:  $(SRCDIR)/%.rc
-	@echo ''; echo Tecmake: compiling $(<F)...
+	@echo ''; echo Tecmake: compiling $(<F) ...
 	$(ECHO)$(RCC) $(RCFLAGS) -O coff -o $@ $<
 
 $(LOHDIR)/%.loh:  $(OBJROOT)/%.lo
-	@echo ''; echo Tecmake: generating $(<F)...
+	@echo ''; echo Tecmake: generating $(<F) ...
 	$(ECHO)$(BIN2C) $< > $@
 
 $(OBJROOT)/%$(LO_SUFFIX).lo:  $(SRCLUADIR)/%.lua
-	@echo ''; echo Tecmake: compiling $(<F)...
+	@echo ''; echo Tecmake: compiling $(<F) ...
 	$(ECHO)$(LUAC) -o $@ $<
 
 ifdef LOHPACK
 $(LOHDIR)/$(LOHPACK):  $(SRCLUA)
-	@echo ''; echo Tecmake: generating $(<F)...
+	@echo ''; echo Tecmake: generating $(<F) ...
 	$(ECHO)$(LUABIN) $(LUAPRE) $(LUAPREFLAGS) -l $(SRCLUADIR) -o $@ $(SRCLUA)
 endif
 
-  
+
 #---------------------------------#
 # Dependencies
 
-# make depend
 #   Build dependencies
 .PHONY: depend
 depend: $(DEPEND)
@@ -1168,7 +1329,7 @@
 	  @echo "" > $(DEPEND)
 	  @which $(CPPC) 2> /dev/null 1>&2 ;\
 	  if [ $$? -eq 0 ]; then \
-	    echo "Tecmake: Building Dependencies... (can be slow)" ;\
+	    echo "Tecmake: Building Dependencies ... [ $(DEPEND) ] (can be slow)" ;\
 	    $(CPPC) $(DEPINCS) $(DEFINES) $(STDDEFS) -MM $(SOURCES) | \
 	    sed -e '1,$$s/^\([^ ]\)/$$(OBJDIR)\/\1/' > $(DEPEND) ;\
 	  else \
@@ -1190,57 +1351,47 @@
 #---------------------------------#
 # Management Rules
 
-# make clean-extra
 #   Remove extra files
 .PHONY: clean-extra
 clean-extra:
 	rm -f $(DEPEND) $(SRELEASE) so_locations
-	
-# make clean-lohs
+
 #   Remove Lua object inclusion files
 .PHONY: clean-lohs
 clean-lohs:
 	rm -f $(LOS) $(LOHS)
-	
-# make clean-obj
+
 #   Remove object files
 .PHONY: clean-obj
 clean-obj:
 	rm -f $(OBJS)
 
-# make clean-target
 #   Remove target
 .PHONY: clean-target
 clean-target:
 	rm -f $(TARGET)
 
-# make clean-dir
 .PHONY: clean-dir
 clean-dir:
 	rm -fr $(OBJROOT) $(TARGETROOT)
 
-# make clean
 #   Remove target and object files
 .PHONY: clean
 clean: clean-target clean-obj
 
-# make strip
 #   Remove symbols from executables
 .PHONY: strip
 strip:
 	test -r $(TARGETDIR)/$(TARGETAPPNAME) && strip $(TARGETDIR)/$(TARGETAPPNAME)
 
-# make rebuild
-#   Rebuild target and object files 
+#   Rebuild target and object files
 .PHONY: rebuild
 rebuild: clean-extra clean-lohs clean-obj clean-target tecmake
 
-# make relink
-#   Rebuild target without rebuilding object files 
+#   Rebuild target without rebuilding object files
 .PHONY: relink
 relink: clean-target tecmake
 
-# make clean-all-obj
 #   Remove target and object files
 .PHONY: clean-all-obj
 clean-all-obj:
@@ -1248,14 +1399,13 @@
 	  (cd $(OBJROOT)/$$d; echo $(OBJ) | xargs rm -f) ;\
 	done
 
-# make clean-all-target
 #   Remove libraries and executables for all platforms
 .PHONY: clean-all-target
 clean-all-target:
 	@for d in $(UNAMES); do \
 	  (rm -f $(TARGETROOT)/$$d/$(TARGETNAME) $(TARGETROOT)/$$d/$(TARGETSLIBNAME) $(TARGETROOT)/$$d/$(TARGETDLIBNAME)) ;\
 	done
- 
+
 #---------------------------------#
 # Remote build
 # There must be aliases in DNS for the known UNAMES
