diff -ru --strip-trailing-cr iup.orig/src/win/iupwin_key.c iup/src/win/iupwin_key.c
--- iup.orig/src/win/iupwin_key.c	2013-08-11 01:09:04.000000000 +0200
+++ iup/src/win/iupwin_key.c	2013-08-18 21:30:43.539964000 +0200
@@ -21,6 +21,11 @@
 #include "iupwin_drv.h"
                    
                    
+/* some older mingw-w64.sf.net headers are missing the following define */
+#ifndef MAPVK_VK_TO_CHAR
+#define MAPVK_VK_TO_CHAR    2
+#endif
+
 typedef struct _Iwin2iupkey
 {
   int iupcode;
