diff -ru --strip-trailing-cr iup.orig/srcmglplot/mgl/mgl_define.h iup/srcmglplot/mgl/mgl_define.h
--- iup.orig/srcmglplot/mgl/mgl_define.h	2012-08-01 13:07:19 +0000
+++ iup/srcmglplot/mgl/mgl_define.h	2012-08-01 13:40:35 +0000
@@ -53,7 +53,7 @@
 #endif
 //-----------------------------------------------------------------------------
 //#ifdef WIN32
-#ifdef _MSC_VER
+#if defined(_MSC_VER) || defined(__MINGW64_VERSION_MAJOR)
 #define mglprintf    _snwprintf
 #else
 #define mglprintf    swprintf
