diff -ru --strip-trailing-cr iup.orig/src/config.mak iup/src/config.mak
--- iup.orig/src/config.mak	2010-11-12 01:41:35.000000000 +0100
+++ iup/src/config.mak	2010-11-16 07:25:17.255943500 +0100
@@ -56,7 +56,7 @@
            gtk/iupgtk_tabs.c gtk/iupgtk_menu.c gtk/iupgtk_list.c gtk/iupgtk_tree.c
            
   ifneq ($(findstring Win, $(TEC_SYSNAME)), )
-    DEFINES += _WIN32_WINNT=0x0500 _WIN32_IE=0x0500 WINVER=0x0500 NOTREEVIEW
+    DEFINES += _WIN32_WINNT=0x0501 _WIN32_IE=0x0501 WINVER=0x0501 NOTREEVIEW
     SRC += win/iupwindows_main.c win/iupwindows_help.c win/iupwindows_info.c
   else
     ifdef GTK_MAC
@@ -99,7 +99,7 @@
   SRC += win/iupwindows_main.c win/iupwindows_help.c win/iupwindows_info.c
 
   INCLUDES += win
-  DEFINES += _WIN32_WINNT=0x0500 _WIN32_IE=0x0500 WINVER=0x0500 NOTREEVIEW
+  DEFINES += _WIN32_WINNT=0x0501 _WIN32_IE=0x0501 WINVER=0x0501 NOTREEVIEW
 endif
 endif
 
diff -ru --strip-trailing-cr iup.orig/srctuio/iup_tuio.cpp iup/srctuio/iup_tuio.cpp
--- iup.orig/srctuio/iup_tuio.cpp	2010-07-22 21:00:13.000000000 +0200
+++ iup/srctuio/iup_tuio.cpp	2010-11-19 12:18:54.751318000 +0100
@@ -377,7 +377,7 @@
 {
   int port = 3333;
   if (params && params[0])
-    port = (int)(long)(params[0]); /* must cast to long first to avoid 64bit compiler error */
+    port = *(int*)params[0];
   
   ih->data = iupALLOCCTRLDATA();
   
@@ -398,7 +398,7 @@
 Ihandle* IupTuioClient(int port)
 {
   void *params[2];
-  params[0] = (void*)port;
+  params[0] = &port;
   params[1] = NULL;
   return IupCreatev("tuioclient", params);
 }
