diff -ru iup.orig/Makefile iup/Makefile
--- iup.orig/Makefile	2014-10-22 18:51:01.094357900 +0200
+++ iup/Makefile	2014-10-23 07:11:30.432052900 +0200
@@ -2,8 +2,8 @@
   WINLIBS = iupole
 endif
 
-.PHONY: do_all iup iupgtk iupmot iupcd iupcontrols iupgl iupglcontrols iup_pplot iup_mglplot iup_scintilla iupim iupimglib ledc iupview iuplua3 iuplua5 iupconsole iupole iupweb iuptuio
-do_all: iup iupcd iupcontrols iupgl iupglcontrols iup_pplot iup_mglplot iup_scintilla iupim iupimglib $(WINLIBS) iupweb iuptuio ledc iupview iuplua5 iupconsole
+.PHONY: do_all iup iupgtk iupmot iupcd iupcontrols iupgl iupglcontrols iupmatrixex iup_pplot iup_mglplot iup_scintilla iupim iupimglib ledc iupview iuplua3 iuplua5 iupconsole iupole iupweb iuptuio
+do_all: iup iupcd iupcontrols iupgl iupglcontrols iupmatrixex iup_pplot iup_mglplot iup_scintilla iupim iupimglib $(WINLIBS) iupweb iuptuio ledc iupview iuplua5 iupconsole
 
 iup iupgtk iupmot:
 	@$(MAKE) --no-print-directory -C ./src/ $@
@@ -11,6 +11,8 @@
 	@$(MAKE) --no-print-directory -C ./srccd/
 iupcontrols:
 	@$(MAKE) --no-print-directory -C ./srccontrols/
+iupmatrixex:
+	@$(MAKE) --no-print-directory -C ./srcmatrixex/
 iup_pplot:
 	@$(MAKE) --no-print-directory -C ./srcpplot/
 iup_mglplot:
